webui/react/src/pages/F_ExpList/glide-table/menu.tsx

```diff
@@ -23,37 +23,38 @@ function useOutsideClickHandler(ref: MutableRefObject<any>, handler: () => void)
   }, [ref, handler]);
 }
 
-export interface TableActionMenuProps extends MenuProps {
-  x: number;
-  y: number;
+export interface TableActionMenuProps {
+  bounds: Rectangle;
   open: boolean;
   handleClose: () => void;
+  items?: MenuItem[];
 }
 
 export const TableActionMenu: React.FC<TableActionMenuProps> = ({
-  x,
-  y,
+  bounds,
   open,
   handleClose,
   items,
 }) => {
-  const menuWidth = 220;
-  const containerRef = useRef(null);
-  useOutsideClickHandler(containerRef, handleClose);
-  const { width } = useResize();
-
+  const divRef = useRef<HTMLDivElement | null>(null);
+  useOutsideClickHandler(divRef, handleClose);
   return (
-    <div
-      ref={containerRef}
-      style={{
-        border: 'solid 1px gray',
-        display: !open ? 'none' : undefined,
-        left: width - x < menuWidth ? width - menuWidth : x,
-        position: 'fixed',
-        top: y,
-        width: menuWidth,
-      }}>
-      <Menu items={items} selectable={false} />
-    </div>
+    <Dropdown menu={items} open={open} overlayStyle={{ minWidth: 'auto' }} placement="bottomLeft">
```

Comment:
we've been avoiding passing style objects to UI kit components, could we change that here?

Reply:
i don't think it's fully avoidable? the dropdown menu takes on the width of the target, so for wide columns, the menu will appear stretched. We could turn this option into a `autoWidthOverlay` prop, but this feels like one-off functionality.

Reply:
imo a prop makes sense, @EmilyBonar @ClaireNeveu thoughts?

Reply:
Identifying what the semantic behavior needed here is and turning that into some kind of option for the Dropdown component is the correct approach.

Reply:
Done!